### PR TITLE
Local models and other modification

### PR DESCRIPTION
**Pull request by Tesla2000**
I am open to a discusion about some changes I added. Especially controversial can be removal of context length check which allows more flesibility in model choice and replacement of setting from config file with default which in my opinion makes it much more intuitive to use. It anything I would be pleased if you merge in change from repo_agent/utils/gitignore_checker.py which I believe is a bugfix. Previous behaviour caused indexing of gitignore files including .venv which shouldn't be the case

### CODE REVIEW

File: repo_agent/utils/gitignore_checker.py [hunk 1]
```diff
@@ -14,6 +16,7 @@ def __init__(self, directory: str, gitignore_path: str):
         self.directory = directory
         self.gitignore_path = gitignore_path
         self.folder_patterns, self.file_patterns = self._load_gitignore_patterns()
+        self.folder_patterns += setting.project.ignore_list
```
Comment (Umpire2018) [inline repo_agent/utils/gitignore_checker.py:19]: > It anything I would be pleased if you merge in change from repo_agent/utils/gitignore_checker.py which I believe is a bugfix. Previous behaviour caused indexing of gitignore files including .venv which shouldn't be the case.

Can you open a new issue to discuss this problem and provide more details?

**Review by Umpire2018 — CHANGES_REQUESTED** [review]
Overall, this PR addresses the following points:

- **Formatting Changes:** You've done a great job ensuring the files are neatly formatted using a formatting tool. However, I believe that these changes should not be bundled with this PR. According to best practices, a PR should ideally focus on a single concern, and mixing formatting with functional changes can make it harder to review.

- **Introduction of response_generators:** I appreciate your effort to introduce response_generators for unified support of OpenAI and Huggingface models. However, I'm not entirely in favor of this approach. A potentially better solution might be to incorporate the `litellm` library, which could provide more streamlined multi-model support.

- **Env Support for API Configuration:** You've also introduced env support to configure API keys and related information. While this is a thoughtful addition, it doesn't align with the current project's approach of using the `pydantic_setting` library in combination with TOML for configuration. Therefore, I don't think this change should be included.

- **Fix typo:** From `latest_verison_substring` to `latest_version_substring`.

**Comment by Tesla2000** [comment]
You are right about litellm I wasn't aware of this library. I see you already added self.folder_patterns += setting.project.ignore_list. You have one more latest_verison_substring in meta_info_utils.py

**Comment by Tesla2000** [comment]
You may close this PR

**Pull request closed without merging**